website/addons/base/views.py

```diff
@@ -466,20 +501,35 @@ def addon_view_or_download_file_legacy(**kwargs):
 
 @must_be_valid_project
 @must_be_contributor_or_public
-def addon_deleted_file(auth, node, **kwargs):
+def addon_deleted_file(auth, node, error_type='BLAME_PROVIDER', **kwargs):
     """Shows a nice error message to users when they try to view
     a deleted file
     """
     # Allow file_node to be passed in so other views can delegate to this one
-    trashed = kwargs.get('file_node') or TrashedFileNode.load(kwargs.get('trashed_id'))
-    if not trashed:
-        raise HTTPError(httplib.NOT_FOUND, {
-            'message_short': 'Not Found',
-            'message_long': 'This file does not exist'
-        })
-
+    file_node = kwargs.get('file_node') or TrashedFileNode.load(kwargs.get('trashed_id'))
+    deleted_by, deleted_on = None, None
+    if isinstance(file_node, TrashedFileNode):
+        deleted_by = file_node.deleted_by
+        deleted_by_guid = file_node.deleted_by._id
+        deleted_on = file_node.deleted_on.strftime("%c") + ' UTC'
+        error_type = 'BLAME_PROVIDER' if file_node.deleted_by is None else 'FILE_GONE'
+    else:
+        error_type = 'DONT_KNOW'
+    file_path = kwargs.get('path', file_node.path)
+    file_name = file_node.name or os.path.basename(file_path)
+    file_name_title, file_name_ext = os.path.splitext(file_name)
+    provider_full = settings.ADDONS_AVAILABLE_DICT[file_node.provider].full_name
     ret = serialize_node(node, auth, primary=True)
     ret.update(rubeus.collect_addon_assets(node))
+    retError = ERROR_MESSAGES[error_type].format(file_name=markupsafe.escape(file_name),
```

Comment:
Because this is a bit of an unusual error reporting mechanism, could you be sure to test this with a malformed google docs filename of the sort we discussed?

In particular, I want to make sure that special characters aren't being double-escaped (once backend, once frontend)

Due to a PR that landed earlier today, Markupsafe rules recently became far more widespread... so for most certain results, consider pulling from develop. Let me know if I can clarify in person.